native/core/src/execution/planner.rs

```diff
@@ -775,6 +776,49 @@ impl PhysicalPlanner {
 
                 Ok(Arc::new(case_expr))
             }
+            ExprStruct::ArrayRepeat(expr) => {
+                let src_expr =
+                    self.create_expr(expr.left.as_ref().unwrap(), Arc::clone(&input_schema))?;
+                let count_expr =
+                    self.create_expr(expr.right.as_ref().unwrap(), Arc::clone(&input_schema))?;
+                // Cast count_expr from Int32 to Int64 to support df count argument
```

Comment:
https://github.com/apache/datafusion/pull/14236

Currently we are manually casting the count_expr to Int64 datatype.
We can remove this once this PR is merged.